docs/architecture.md

```diff
@@ -0,0 +1,46 @@
+# Gatling Operator Architecture and Design
+
+<!-- TOC -->
+
+- [Gatling Operator Architecture and Design](#gatling-operator-architecture-and-design)
+	- [Architecture Overview](#architecture-overview)
+	- [Gatling Runner Pod as Multi Container Pod](#gatling-runner-pod-as-multi-container-pod)
+
+<!-- /TOC -->
+
+## Architecture Overview
+
+The diagram below shows how Gatling Operator works in conjunction with Gatling Custom Resource (CR).
+Based on Gatling CR, actions in the lifecycle of distributed Gatling load testing such as running load testing, generating reports, sending notification message, and cleaning up the resources are executed by relevant custom controller (`Gatling Controller` in figure below).
```

Comment:
[IMO] I think using brackets for long examples make sentence easier to understand.

```suggestion
Based on Gatling CR, actions in the lifecycle of distributed Gatling load testing (such as running load testing, generating reports, sending notification message, and cleaning up the resources) are executed by relevant custom controller (`Gatling Controller` in figure below).
```

Reply:
@kayamin Thanks for the suggestion. That makes sense!